internal/namespace/manager.go

```diff
@@ -21,271 +21,830 @@ import (
 	"github.com/hyperledger/firefly/pkg/blockchain"
 	"github.com/hyperledger/firefly/pkg/core"
 	"github.com/hyperledger/firefly/pkg/database"
 	"github.com/hyperledger/firefly/pkg/dataexchange"
+	"github.com/hyperledger/firefly/pkg/identity"
 	"github.com/hyperledger/firefly/pkg/sharedstorage"
 	"github.com/hyperledger/firefly/pkg/tokens"
 )
 
+var (
+	blockchainConfig    = config.RootArray("plugins.blockchain")
+	tokensConfig        = config.RootArray("plugins.tokens")
+	databaseConfig      = config.RootArray("plugins.database")
+	sharedstorageConfig = config.RootArray("plugins.sharedstorage")
+	dataexchangeConfig  = config.RootArray("plugins.dataexchange")
+	identityConfig      = config.RootArray("plugins.identity")
+
+	// Deprecated configs
+	deprecatedTokensConfig        = config.RootArray("tokens")
+	deprecatedBlockchainConfig    = config.RootSection("blockchain")
+	deprecatedDatabaseConfig      = config.RootSection("database")
+	deprecatedSharedStorageConfig = config.RootSection("sharedstorage")
+	deprecatedDataexchangeConfig  = config.RootSection("dataexchange")
+)
+
 type Manager interface {
-	// Init initializes the manager
-	Init(ctx context.Context, di database.Plugin) error
-	// GetDefaultKey retrieves the default blockchain key for this namespace
-	GetDefaultKey(ns string) string
-	// GetMultipartyConfig retrieves a config string from the namespace's multiparty config or the (deprecated) root config
-	GetMultipartyConfig(ns string, key config.RootKey) string
+	Init(ctx context.Context, cancelCtx context.CancelFunc) error
+	Start() error
+	WaitStop()
+
+	Orchestrator(ns string) orchestrator.Orchestrator
+	SPIEvents() spievents.Manager
+	GetNamespaces(ctx context.Context) ([]*core.Namespace, error)
+	GetOperationByNamespacedID(ctx context.Context, nsOpID string) (*core.Operation, error)
+	ResolveOperationByNamespacedID(ctx context.Context, nsOpID string, op *core.OperationUpdateDTO) error
+}
+
+type namespace struct {
+	description  string
+	orchestrator orchestrator.Orchestrator
+	config       orchestrator.Config
+	plugins      orchestrator.Plugins
 }
 
 type namespaceManager struct {
-	ctx           context.Context
-	nsConfig      map[string]config.Section
-	bcPlugins     map[string]blockchain.Plugin
-	dbPlugins     map[string]database.Plugin
-	dxPlugins     map[string]dataexchange.Plugin
-	ssPlugins     map[string]sharedstorage.Plugin
-	tokensPlugins map[string]tokens.Plugin
+	ctx         context.Context
+	cancelCtx   context.CancelFunc
+	namespaces  map[string]*namespace
+	pluginNames map[string]bool
+	plugins     struct {
+		blockchain    map[string]blockchainPlugin
+		identity      map[string]identityPlugin
+		database      map[string]databasePlugin
+		sharedstorage map[string]sharedStoragePlugin
+		dataexchange  map[string]dataExchangePlugin
+		tokens        map[string]tokensPlugin
+	}
+	metricsEnabled bool
+	metrics        metrics.Manager
+	adminEvents    spievents.Manager
+	utOrchestrator orchestrator.Orchestrator
```

Comment:
I don't love this testing workaround, but I was struggling to find a better way.